src/lib/viewers/text/CSVViewer.js

```diff
@@ -67,7 +67,18 @@ class CSVViewer extends TextBaseViewer {
                             if (this.isDestroyed() || !results) {
                                 return;
                             }
-                            this.data = results.data;
+
+                            const { errors = [], data } = results;
+
+                            if (errors.length) {
+                                const error = new PreviewError(ERROR_CODE.LOAD_CSV, undefined, {
```

Comment:
We may want to use a different error code to differentiate load errors vs parse errors.